src/config.ts

```diff
@@ -104,24 +114,13 @@ export class CacheConfig {
 
     self.keyEnvs = keyEnvs;
 
-    // Installed packages and their versions are also considered for the key.
-    const packages = await getPackages();
-    hasher.update(packages);
```

Comment:
I believe I understand what this whole problem is all about:

We used to consider all the `cargo install --list` (aka the `.cargo/bin` binaries) for the cache key. And running a `cargo install` as part of the workflow obviously changes that.

Reply:
Correct, any changes via `cargo install` between the main action step (restore) and the post action step would change what's installed and hence re-calculating in post action step cause the key to change.

The original intent of that change was to allow it pin specific additional installs, but that's simply not possible without having distinct action steps, where the user has control over the items captured.

I considered various options but came to the conclusion that the solution proposed by @Boshen to use [taiki-e/install-action](https://github.com/taiki-e/install-action) to install a binary version of udeps instead of trying to get it caching properly was the best solution.